webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js

```diff
@@ -25,91 +25,84 @@ const Disk = ({
-    updateDisk('datastore', { target: { value: null } });
-  };
-  const updateDatastore = newValues => {
-    updateDisk('datastore', newValues);
-    updateDisk('storagePod', { target: { value: null } });
-  };
-
-  return (
-    <div className="disk-container">
-      <div className="form-group">
-        <label className="col-md-2 control-label">{__('Disk name')}</label>
-        <div className="col-md-4">{name}</div>
-        <div className="col-md-2">
-          {!vmExists && (
-            <Button className="close" onClick={removeDisk}>
-              <span aria-hidden="true">&times;</span>
-            </Button>
-          )}
-        </div>
+}) => (
+  <div className="disk-container">
+    <div className="form-group">
+      <label className="col-md-2 control-label">{__('Disk name')}</label>
+      <div className="col-md-4">{name}</div>
+      <div className="col-md-2">
+        {!vmExists && (
+          <Button className="close" onClick={removeDisk}>
+            <span aria-hidden="true">&times;</span>
+          </Button>
+        )}
       </div>
-      {!(datastore && datastore.length) && (
-        <Select
-          label={__('Storage Pod')}
-          value={storagePod}
-          disabled={vmExists}
-          onChange={newValues => updateStoragePod(newValues)}
-          options={storagePods}
-          allowClear
-          key="storagePodsSelect"
-          status={storagePodsStatus}
-          errorMessage={storagePodsError}
-          className="storage-pod"
-        />
-      )}
-      {!(storagePod && storagePod.length) && (
-        <Select
-          disabled={vmExists}
-          label={__('Data store')}
-          value={datastore}
-          onChange={newValues => updateDatastore(newValues)}
-          options={datastores}
-          allowClear
-          key="datastoresSelect"
-          status={datastoresStatus}
-          errorMessage={datastoresError}
-          className="datastore"
-        />
-      )}
-
+    </div>
+    {!(datastore && datastore.length) && (
       <Select
-        label={__('Disk Mode')}
-        value={mode}
+        label={__('Storage Pod')}
+        value={storagePod}
         disabled={vmExists}
-        onChange={newValues => updateDisk('mode', newValues)}
-        options={diskModeTypes}
+        onValueChange={newValue =>
+          updateDisk({ storagePod: newValue, datastore: '' })
+        }
+        options={storagePods}
+        allowClear
+        key="storagePodsSelect"
+        status={storagePodsStatus}
+        errorMessage={storagePodsError}
+        className="storage-pod"
       />
-
-      <NumericInput
-        value={sizeGb}
-        minValue={1}
-        format={v => `${v} GB`}
-        className="text-vmware-size"
-        onChange={newValues => updateDisk('sizeGb', newValues)}
-        label={__('Size (GB)')}
-      />
-
-      <Checkbox
-        label={__('Thin provision')}
-        checked={thin}
+    )}
+    {!(storagePod && storagePod.length) && (
```

Comment:
would be better to use a conditional (ternary) operator:

``` storagePod && storagePod.length ? <Select ......... /> : null```

so you won't return false in React.